experimental/AlgebraicStatistics/src/GaussianGraphicalModels.jl

```diff
@@ -235,167 +300,329 @@ defined by
+of $K$.
 
 ## Examples
 
 ```jldoctest
-julia> M = graphical_model(graph_from_edges([[1,2], [2,3]]), gaussian_ring(3))
-Gaussian graphical model on an undirected graph with edges:
-(1, 2), (2, 3)
+julia> M = gaussian_graphical_model(graph_from_edges([[1,2],[2,3]]))
+Gaussian Graphical Model on a Undirected graph with 3 nodes and 2 edges
 
 julia> parametrization(M)
 Ring homomorphism
   from multivariate polynomial ring in 6 variables over QQ
-  to fraction field of multivariate polynomial ring
+  to localization of multivariate polynomial ring in 5 variables over QQ at products of (-k[2, 1]^2*k[3, 3] - k[3, 2]^2*k[1, 1] + k[1, 1]*k[2, 2]*k[3, 3])
 defined by
-  s[1, 1] -> (k[2, 2]*k[3, 3] - k[2, 3]^2)//(k[1, 1]*k[2, 2]*k[3, 3] - k[1, 1]*k[2, 3]^2 - k[1, 2]^2*k[3, 3])
-  s[1, 2] -> (-k[1, 2]*k[3, 3])//(k[1, 1]*k[2, 2]*k[3, 3] - k[1, 1]*k[2, 3]^2 - k[1, 2]^2*k[3, 3])
-  s[1, 3] -> (k[1, 2]*k[2, 3])//(k[1, 1]*k[2, 2]*k[3, 3] - k[1, 1]*k[2, 3]^2 - k[1, 2]^2*k[3, 3])
-  s[2, 2] -> (k[1, 1]*k[3, 3])//(k[1, 1]*k[2, 2]*k[3, 3] - k[1, 1]*k[2, 3]^2 - k[1, 2]^2*k[3, 3])
-  s[2, 3] -> (-k[1, 1]*k[2, 3])//(k[1, 1]*k[2, 2]*k[3, 3] - k[1, 1]*k[2, 3]^2 - k[1, 2]^2*k[3, 3])
-  s[3, 3] -> (k[1, 1]*k[2, 2] - k[1, 2]^2)//(k[1, 1]*k[2, 2]*k[3, 3] - k[1, 1]*k[2, 3]^2 - k[1, 2]^2*k[3, 3])
+  s[1, 1] -> (k[3, 2]^2 - k[2, 2]*k[3, 3])/(k[2, 1]^2*k[3, 3] + k[3, 2]^2*k[1, 1] - k[1, 1]*k[2, 2]*k[3, 3])
+  s[1, 2] -> k[2, 1]*k[3, 3]/(k[2, 1]^2*k[3, 3] + k[3, 2]^2*k[1, 1] - k[1, 1]*k[2, 2]*k[3, 3])
+  s[1, 3] -> -k[2, 1]*k[3, 2]/(k[2, 1]^2*k[3, 3] + k[3, 2]^2*k[1, 1] - k[1, 1]*k[2, 2]*k[3, 3])
+  s[2, 2] -> -k[1, 1]*k[3, 3]/(k[2, 1]^2*k[3, 3] + k[3, 2]^2*k[1, 1] - k[1, 1]*k[2, 2]*k[3, 3])
+  s[2, 3] -> k[3, 2]*k[1, 1]/(k[2, 1]^2*k[3, 3] + k[3, 2]^2*k[1, 1] - k[1, 1]*k[2, 2]*k[3, 3])
+  s[3, 3] -> (k[2, 1]^2 - k[1, 1]*k[2, 2])/(k[2, 1]^2*k[3, 3] + k[3, 2]^2*k[1, 1] - k[1, 1]*k[2, 2]*k[3, 3])
 ```
 """
-function parametrization(M::GraphicalModel{Graph{Undirected}, GaussianRing})
+function parametrization(M::GaussianGraphicalModel{Graph{Undirected}, T}) where T
   G = graph(M)
-  S = ring(M)
-  R = param_ring(M)
+  S, _ = model_ring(M)
+  R, _ = parameter_ring(M)
   K = concentration_matrix(M)
+  Rloc, iota = localization(R, powers_of_element(det(K)))
   adj = adjugate(K)
-  hom(ring(S), fraction_field(R), reduce(vcat, [[adj[i,j]//det(K) for j in i:n_vertices(G)] for i in 1:n_vertices(G)]))
+  hom(S, Rloc, reduce(vcat, [[iota(adj[i,j])/iota(det(K)) for j in i:n_vertices(G)] for i in 1:n_vertices(G)]))
+end
+
+@doc raw"""
+    ci_structure(M::GaussianGraphicalModel{Graph{Undirected}, T})
+
+Return the set of elementary CI statements which are satisfied by every distribution in the
+graphical model. This is the same as the `global_markov` property of the underlying graph.
+"""
+function ci_structure(M::GaussianGraphicalModel{Graph{Undirected}, T}) where T
+  global_markov(graph(M))
```

Comment:
We use explicit `return` except for onelines.
```suggestion
  return global_markov(graph(M))
```